docker/launch-container.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+set -e
+
+# PyHSS Docker Launch Script
+# This script processes the configuration template and starts the specified service
+
+# Default values
+TEMPLATE_FILE="${CONFIG_TEMPLATE:-/opt/pyhss/config.yaml.template}"
+PYHSS_CONFIG="${PYHSS_CONFIG:-/opt/pyhss/config.yaml}"
+CONTAINER_ROLE="${CONTAINER_ROLE:-undefined}"
+
+echo "=== PyHSS Container Startup ==="
+echo "Template file: $TEMPLATE_FILE"
+echo "Config file: $PYHSS_CONFIG"
+
+# Generate configuration file from template
+envsubst < "$TEMPLATE_FILE" > "$PYHSS_CONFIG"
+
+
+if [ $? -eq 0 ]; then
+    echo "Configuration file generated successfully: $PYHSS_CONFIG"
+else
+    echo "ERROR: Failed to generate configuration file"
+    exit 1
+fi
```

Comment:
This doesn't work as expected: as `set -e` is used (which is good!), the script already aborts in the envsubst line if the command fails.

```suggestion
if ! envsubst < "$TEMPLATE_FILE" > "$PYHSS_CONFIG"; then
    echo "ERROR: Failed to generate configuration file"
    exit 1
fi

echo "Configuration file generated successfully: $PYHSS_CONFIG"
```